### PR TITLE
VALUES support for query builder

### LINKED ISSUE #211: Building SPARQL with VALUES clause
On the Basic Tutorial section **UserGuide Building SPARQL**, there is no indication of how to create a VALUES clause within a SPARQL. For example, 

`SELECT ?x
 WHERE { VALUES (?x) { (http://www.w3.org/2001/vcard-rdf/3.0#FN) } }`

How can I create a SPARQL with a VALUES clause using the `QueryBuilder`?

### PR DESCRIPTION
**Pull request by tpluscode**
WIP adding `VALUES` to query builder API:

* [x] adding `VALUES` to graph patterns
  * [x] tagged and typed literals
  * [x] URIs
* [x] adding `VALUES` to entire query
* [x] XML doc

fixes #211

### CODE REVIEW

File: Libraries/dotNetRDF/Query/Builder/QueryBuilder.cs [hunk 1]
```diff
@@ -377,10 +384,7 @@ private void BuildRootGraphPattern(SparqlQuery query)
         {
             var rootGraphPattern = RootGraphPatternBuilder.BuildGraphPattern(Prefixes);
 
-            if (!rootGraphPattern.IsEmpty)
-            {
-                query.RootGraphPattern = rootGraphPattern;
-            }
+            query.RootGraphPattern = rootGraphPattern;
```
Comment (tpluscode) [inline Libraries/dotNetRDF/Query/Builder/QueryBuilder.cs]: This one change seems unrelated but I found that `VALUES` to not switch `IsEmpty`. 

Does is make any difference if the root in null or not?

File: Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs [hunk 2]
```diff
@@ -33,58 +33,37 @@ internal static class LiteralExpressionExtensions
     {
         private static readonly NodeFactory NodeFactory = new NodeFactory();
 
-        private static ILiteralNode ToLiteral<T>(T numericValue)
+        internal static ILiteralNode ToLiteral(object value)
         {
-            if (typeof(T) == typeof(int))
+            switch (value)
```
Comment (tpluscode) [inline Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs]: Turned those ifs into C# 7.1 pattern matching of generic type argument. Do you have any objections? I understand that it only affect build-time requirements?

File: Libraries/dotNetRDF/Query/Builder/IQueryBuilder.cs [hunk 3]
```diff
@@ -41,8 +41,13 @@ public interface IQueryBuilder
         /// <summary>
         /// Gets the builder associated with the root graph pattern.
         /// </summary>
+        [Obsolete("Please use Root property")]
         GraphPatternBuilder RootGraphPatternBuilder { get; }
         /// <summary>
+        /// Gets the builder associated with the root graph pattern.
+        /// </summary>
+        IGraphPatternBuilder Root { get; }
```
Comment (tpluscode) [inline Libraries/dotNetRDF/Query/Builder/IQueryBuilder.cs]: Added a new alias and made the long property obsolete bc it simply seems overly verbose

File: Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs (hunk 2 above)
Comment (kal) [reply Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs]: No objection from me - I tend to update C# syntax as I go along in any case.

File: Libraries/dotNetRDF/Query/Builder/QueryBuilder.cs (hunk 1 above)
Comment (kal) [reply Libraries/dotNetRDF/Query/Builder/QueryBuilder.cs]: I think if the root graph pattern is null / empty then you should just end up with the values in the result set. Not 100% sure, but I don't think this is a problem ;-)

File: Libraries/dotNetRDF/Query/Builder/IQueryBuilder.cs (hunk 3 above)
Comment (kal) [reply Libraries/dotNetRDF/Query/Builder/IQueryBuilder.cs]: 👍

**Comment by kal** [comment]
This is looking really good!

**Comment by tpluscode** [comment]
Added XML comments. Please check my latest commit and we're ready to merge

**Comment by kal** [comment]
Thanks @tpluscode!

**Comment by assumption** [comment]
Thank you @tpluscode for this!

**Pull request merged**